src/Core/Grid/Definition/Factory/LinkBlockDefinitionFactory.php

```diff
@@ -0,0 +1,135 @@
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\AbstractGridDefinitionFactory;
+
+/**
+ * Class LinkBlockDefinitionFactory
+ * @package PrestaShop\Module\LinkList\Core\Grid\Definition\Factory
+ */
+final class LinkBlockDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    const FACTORY_ID = 'link_widget_grid_';
+
+    /**
+     * @var array
+     */
+    private $hook;
+
+    /**
+     * LinkBlockDefinitionFactory constructor.
+     * @param array $hook
+     */
+    public function __construct(array $hook)
+    {
+        $this->hook = $hook;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function getId()
+    {
+        return self::FACTORY_ID . $this->hook['id_hook'];
+    }
+
+
+    /**
+     * @inheritDoc
+     */
+    protected function getName()
+    {
+        return $this->hook['name'] . ' ' . $this->hook['title'];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function getColumns()
+    {
+        return (new ColumnCollection())
+            ->add((new DataColumn('id_link_block'))
+                ->setName($this->trans('ID', [], 'Modules.Linklist.Admin'))
+                ->setOptions([
+                    'field' => 'id_hook'
+                ])
+            )
+            ->add((new DataColumn('block_name'))
+                ->setName($this->trans('Name of the block', [], 'Modules.Linklist.Admin'))
+                ->setOptions([
+                    'field' => 'block_name'
+                ])
+            )
+            ->add((new PositionColumn('position'))
+                ->setName($this->trans('Position', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'position',
+                    'id_field' => 'id_link_block',
+                    'id_parent_field' => 'id_hook',
+                    'position_update_route' => 'admin_link_block_update_positions',
+                ])
+            )
+            ->add((new ActionColumn('actions'))
+                ->setOptions([
+                    'actions' => (new RowActionCollection())
+                        ->add((new LinkRowAction('edit'))
+                            ->setName($this->trans('Edit', [], 'Admin.Global'))
+                            ->setIcon('edit')
+                            ->setOptions([
+                                'route' => 'admin_link_block_edit',
+                                'route_param_name' => 'linkBlockId',
+                                'route_param_field' => 'id_link_block',
+                            ])
+                        )
+                        ->add((new SubmitRowAction('delete'))
+                            ->setName($this->trans('Delete', [], 'Admin.Actions'))
+                            ->setIcon('delete')
+                            ->setOptions([
+                                'method' => 'DELETE',
```

Comment:
Regarding some recent issues, it seems using `DELETE` does not work everywhere. You should use `POST` just in case.